File: Sample/BasicSample/sample_cpp/PostEffect/PostEffect_GrayScale.cpp
#include <Altseed.h>

// レイヤーにグレースケールのポストエフェクトを適用する。
void PostEffect_GrayScale()
{
	// Altseedを初期化する
	asd::Engine::Initialize(asd::ToAString("PostEffect_GrayScale").c_str(), 640, 480, asd::EngineOption());

	auto texture = asd::Engine::GetGraphics()->CreateTexture2D(asd::ToAString("Data/Texture/Picture1.png").c_str());

	// シーンのコンストラクタを呼び出す。
	auto scene = std::make_shared<asd::Scene>();

	// レイヤーのコンストラクタを呼び出す。
	auto layer = std::make_shared<asd::Layer2D>();

	// オブジェクトのコンストラクタを呼び出す。
	auto obj = std::make_shared<asd::TextureObject2D>();

	// オブジェクトの位置とテクスチャを設定。
	obj->SetPosition(asd::Vector2DF(50, 50));
	obj->SetTexture(texture);

	// シーンを変更し、そのシーンにレイヤーを追加し、そのレイヤーにオブジェクトを追加する。
	asd::Engine::ChangeScene(scene);
	scene->AddLayer(layer);
	layer->AddObject(obj);

	// レイヤーにグレースケールのポストエフェクトを適用。
	layer->AddPostEffect(std::make_shared<asd::PostEffectGrayScale>());

	while (asd::Engine::DoEvents())
	{
		asd::Engine::Update();
	}

	asd::Engine::Terminate();
}

File: Sample/BasicSample/sample_cs/PostEffect/PostEffect_GrayScale.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sample_cs
{
    class PostEffect_GrayScale : ISample
    {
        public string Description
        {
            get { return ""; }
        }

        // レイヤーにグレースケールのポストエフェクトを適用する。
        public void Run()
        {
            // Altseedを初期化する
            asd.Engine.Initialize("PostEffect_GrayScale", 640, 480, new asd.EngineOption());
            
            // 画像を読み込む。
            var texture = asd.Engine.Graphics.CreateTexture2D("Data/Texture/Picture1.png");

            // シーンクラスのインスタンスを生成する。
            var scene = new asd.Scene();

            // レイヤークラスのインスタンスを生成する。
            var layer = new asd.Layer2D();

            // 画像描画オブジェクトのインスタンスを生成する。
            var obj = new asd.TextureObject2D();

            // オブジェクトの位置とテクスチャを設定する。
            obj.Position = new asd.Vector2DF(50, 50);
            obj.Texture = texture;

            // シーンを変更し、そのシーンにレイヤーを追加し、そのレイヤーにオブジェクトを追加する。
            asd.Engine.ChangeScene(scene);
            scene.AddLayer(layer);
            layer.AddObject(obj);

            // レイヤーにグレースケールのポストエフェクトを適用する。
            layer.AddPostEffect(new asd.PostEffectGrayScale());
            
            // Altseedのウインドウが閉じられていないか確認する。
            while (asd.Engine.DoEvents())
            {
                // Altseedを更新する。
                asd.Engine.Update();
            }
            
            // Altseedを終了する。
            asd.Engine.Terminate();
        }
    }
}
